impeller/renderer/backend/gles/buffer_bindings_gles.cc

```diff
@@ -153,11 +153,17 @@ bool BufferBindingsGLES::ReadUniformsBindings(const ProcTableGLES& gl,
 
 bool BufferBindingsGLES::BindVertexAttributes(const ProcTableGLES& gl,
                                               size_t binding,
-                                              size_t vertex_offset) const {
+                                              size_t vertex_offset) {
   if (binding >= vertex_attrib_arrays_.size()) {
     return false;
   }
 
+  if (!gl.GetCapabilities()->IsES()) {
```

Comment:
I think any non-ES GL will need VAOs? FYI @chinmaygarde

Reply:
I'd put this behind a cap check. Because vertex arrays are available on desktop GL, but also ES 3.0 and above or ES 2.0 with the `GL_OES_vertex_array_object` extension. This extension is almost always available. This is a fine start for now though. But file a followup?

Reply:
This may be because you just wanted to make things work on desktop GL. So feel free to ignore if we are going to patch this in later.

But, VAOs are an optimization that actually become a pessimization as used here.

VAOs encode the state that is configured by the `EnableVertexAttribArray`/`VertexAttribPointer` goop later into a single object that can be bound. So, you only need to configure the arrays once (ideally during "pipeline" construction), save the VAO, and bind it before the draw call while skipping the rest of the API traffic.

Creating the new VAO per call is just extra ceremony that is unnecessary.

I'd create a separate object with its own entry in the reactor handles table that is configured when the pipeline is created and a bind call that either binds the VAO if one is available or dumps additional traffic if not.

But, we can do this all later. This still LGTM and will work to unblock desktop GL.

Reply:
Yeah the goal is very much "make it render" for now. I don't know if we'll get to the point where we work on GL performance. I'd like to think we can push folks to switch to Vulkan where feasible. But if that doesn't happen then we'd have some work to do to write a more 3.0-style renderer.